In [1]:
import pandas as pd

In [4]:
df=pd.read_csv(r'C:\Users\dell\Downloads\sales_data_final.csv')

In [5]:
df.head(3)

order_id order_date ship_date       ship_mode    customer_name  \
0   AG-2011-2040   1/1/2011  1/6/2011  Standard Class  Toby Braunhardt   
1  IN-2011-47883   1/1/2011  1/8/2011  Standard Class      Joseph Holt   
2   HU-2011-1220   1/1/2011  1/5/2011    Second Class    Annie Thurman   

    segment            state    country  market   region  ...  \
0  Consumer      Constantine    Algeria  Africa   Africa  ...   
1  Consumer  New South Wales  Australia    APAC  Oceania  ...   
2  Consumer         Budapest    Hungary    EMEA     EMEA  ...   

          category sub_category              product_name sales quantity  \
0  Office Supplies      Storage       Tenex Lockers, Blue   408        2   
1  Office Supplies     Supplies  Acme Trimmer, High Speed   120        3   
2  Office Supplies      Storage   Tenex Box, Single Width    66        4   

   discount   profit  shipping_cost  order_priority  year  
0       0.0  106.140          35.46          Medium  2011  
1       0.1   36.036           9.72          Medium  2011  
2       0.0   29.640           8.17            High  2011  

[3 rows x 21 columns]

In [9]:
df.columns# List of Columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'quantity',
       'discount', 'profit', 'shipping_cost', 'order_priority', 'year'],
      dtype='object')

In [7]:
df['order_date']# Series of Partiular Column

0          1/1/2011
1          1/1/2011
2          1/1/2011
3          1/1/2011
4          1/1/2011
            ...    
51285    12/31/2014
51286    12/31/2014
51287    12/31/2014
51288    12/31/2014
51289    12/31/2014
Name: order_date, Length: 51290, dtype: object

In [8]:
type(df['order_date'])# Type Series

pandas.core.series.Series

In [10]:
df[['order_date','country']]# Selected Columns but Datatype will be DataFrame

order_date        country
0        1/1/2011        Algeria
1        1/1/2011      Australia
2        1/1/2011        Hungary
3        1/1/2011         Sweden
4        1/1/2011      Australia
...           ...            ...
51285  12/31/2014  United States
51286  12/31/2014        Morocco
51287  12/31/2014      Nicaragua
51288  12/31/2014         Mexico
51289  12/31/2014  United States

[51290 rows x 2 columns]

In [12]:
df.dtypes# Every Column it wil be give the data type

order_id           object
order_date         object
ship_date          object
ship_mode          object
customer_name      object
segment            object
state              object
country            object
market             object
region             object
product_id         object
category           object
sub_category       object
product_name       object
sales              object
quantity            int64
discount          float64
profit            float64
shipping_cost     float64
order_priority     object
year                int64
dtype: object

In [13]:
df['profit']

0        106.1400
1         36.0360
2         29.6400
3        -26.0550
4         37.7700
           ...   
51285      4.5188
51286      0.4200
51287     12.3600
51288      0.5600
51289     -0.6048
Name: profit, Length: 51290, dtype: float64

In [17]:
df['profit','Year']# it will give te error if more then two column we need to enter them in list


KeyError: ('profit', 'Year')

In [16]:
df[['profit','year']]

profit  year
0      106.1400  2011
1       36.0360  2011
2       29.6400  2011
3      -26.0550  2011
4       37.7700  2011
...         ...   ...
51285    4.5188  2014
51286    0.4200  2014
51287   12.3600  2014
51288    0.5600  2014
51289   -0.6048  2014

[51290 rows x 2 columns]

In [18]:
type(df[['profit','year']])

pandas.core.frame.DataFrame

In [19]:
df.columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'quantity',
       'discount', 'profit', 'shipping_cost', 'order_priority', 'year'],
      dtype='object')

In [21]:
df[['order_id','product_name','quantity']]

order_id                                     product_name  \
0         AG-2011-2040                              Tenex Lockers, Blue   
1        IN-2011-47883                         Acme Trimmer, High Speed   
2         HU-2011-1220                          Tenex Box, Single Width   
3      IT-2011-3647632                      Enermax Note Cards, Premium   
4        IN-2011-47883                       Eldon Light Bulb, Duo Pack   
...                ...                                              ...   
51285   CA-2014-115427  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl   
51286     MO-2014-2560          Wilson Jones Hole Reinforcements, Clear   
51287   MX-2014-110527           Hon Color Coded Labels, 5000 Label Set   
51288   MX-2014-114783           Hon Legal Exhibit Labels, Alphabetical   
51289   CA-2014-156720                              Bagged Rubber Bands   

       quantity  
0             2  
1             3  
2             4  
3             3  
4             5  
...         ...  
51285         2  
51286         1  
51287         3  
51288         1  
51289         3  

[51290 rows x 3 columns]

In [23]:
df.describe()# only for numeric columns

quantity      discount        profit  shipping_cost          year
count  51290.000000  51290.000000  51290.000000   51290.000000  51290.000000
mean       3.476545      0.142908     28.641740      26.375915   2012.777208
std        2.278766      0.212280    174.424113      57.296804      1.098931
min        1.000000      0.000000  -6599.978000       0.000000   2011.000000
25%        2.000000      0.000000      0.000000       2.610000   2012.000000
50%        3.000000      0.000000      9.240000       7.790000   2013.000000
75%        5.000000      0.200000     36.810000      24.450000   2014.000000
max       14.000000      0.850000   8399.976000     933.570000   2014.000000

In [26]:
# Filter only tht Data wheere data type is object
df.dtypes=='object'


order_id           True
order_date         True
ship_date          True
ship_mode          True
customer_name      True
segment            True
state              True
country            True
market             True
region             True
product_id         True
category           True
sub_category       True
product_name       True
sales              True
quantity          False
discount          False
profit            False
shipping_cost     False
order_priority     True
year              False
dtype: bool

In [27]:
# Filter only tht Data wheere data type is object
df.dtypes[df.dtypes=='object']


order_id          object
order_date        object
ship_date         object
ship_mode         object
customer_name     object
segment           object
state             object
country           object
market            object
region            object
product_id        object
category          object
sub_category      object
product_name      object
sales             object
order_priority    object
dtype: object

In [28]:
# Filter only tht Data wheere data type is object
df.dtypes[df.dtypes=='object'].index

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'order_priority'],
      dtype='object')

In [30]:
# Filter only tht Data wheere data type is object
df=df[df.dtypes[df.dtypes=='object'].index]

In [31]:
df.describe()

order_id order_date   ship_date       ship_mode  \
count            51290      51290       51290           51290   
unique           25035       1430        1464               4   
top     CA-2014-100111  6/18/2014  11/22/2014  Standard Class   
freq                14        135         130           30775   

          customer_name   segment       state        country market   region  \
count             51290     51290       51290          51290  51290    51290   
unique              795         3        1094            147      7       13   
top     Muhammed Yedwab  Consumer  California  United States   APAC  Central   
freq                108     26518        2001           9994  11002    11117   

             product_id         category sub_category product_name  sales  \
count             51290            51290        51290        51290  51290   
unique            10292                3           17         3788   2246   
top     OFF-AR-10003651  Office Supplies      Binders      Staples     13   
freq                 35            31273         6152          227    589   

       order_priority  
count           51290  
unique              4  
top            Medium  
freq            29433

In [35]:
df=pd.read_csv(r'C:\Users\dell\Downloads\sales_data_final.csv')


In [36]:
df['order_id'][1:40:3]

1       IN-2011-47883
4       IN-2011-47883
7       IN-2011-79397
10      IN-2011-65159
13      IN-2011-33652
16        IR-2011-770
19      ID-2011-12596
22        IR-2011-770
25      IN-2011-65159
28     MX-2011-140641
31     MX-2011-111255
34     MX-2011-159373
37    IT-2011-2942451
Name: order_id, dtype: object

In [38]:
df['category']

0        Office Supplies
1        Office Supplies
2        Office Supplies
3        Office Supplies
4              Furniture
              ...       
51285    Office Supplies
51286    Office Supplies
51287    Office Supplies
51288    Office Supplies
51289    Office Supplies
Name: category, Length: 51290, dtype: object

In [39]:
df['category1']

KeyError: 'category1'

In [40]:
df['category1']='Sudh' # New Column is Created

In [41]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... sub_category  \
0       Africa  ...      Storage   
1      Oceania  ...     Supplies   
2         EMEA  ...      Storage   
3        North  ...        Paper   
4      Oceania  ...  Furnishings   
...        ...  ...          ...   
51285     West  ...      Binders   
51286   Africa  ...      Binders   
51287  Central  ...       Labels   
51288    North  ...       Labels   
51289     West  ...    Fasteners   

                                          product_name sales quantity  \
0                                  Tenex Lockers, Blue   408        2   
1                             Acme Trimmer, High Speed   120        3   
2                              Tenex Box, Single Width    66        4   
3                          Enermax Note Cards, Premium    45        3   
4                           Eldon Light Bulb, Duo Pack   114        5   
...                                                ...   ...      ...   
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

      discount    profit  shipping_cost  order_priority  year category1  
0          0.0  106.1400          35.46          Medium  2011      Sudh  
1          0.1   36.0360           9.72          Medium  2011      Sudh  
2          0.0   29.6400           8.17            High  2011      Sudh  
3          0.5  -26.0550           4.82            High  2011      Sudh  
4          0.1   37.7700           4.70          Medium  2011      Sudh  
...        ...       ...            ...             ...   ...       ...  
51285      0.2    4.5188           0.89          Medium  2014      Sudh  
51286      0.0    0.4200           0.49          Medium  2014      Sudh  
51287      0.0   12.3600           0.35          Medium  2014      Sudh  
51288      0.0    0.5600           0.20          Medium  2014      Sudh  
51289      0.2   -0.6048           0.17          Medium  2014      Sudh  

[51290 rows x 22 columns]

In [42]:
df.columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'quantity',
       'discount', 'profit', 'shipping_cost', 'order_priority', 'year',
       'category1'],
      dtype='object')

In [44]:
df['order_id'].isnull()# it will check whether Column has Null Value or not

0        False
1        False
2        False
3        False
4        False
         ...  
51285    False
51286    False
51287    False
51288    False
51289    False
Name: order_id, Length: 51290, dtype: bool

In [47]:
df1=pd.read_csv(r'C:\Users\dell\FSDS\DressSales.csv')

In [48]:
 df1.columns

Index(['Dress_ID', '29/8/2013', '31/8/2013', '2013-02-09 00:00:00',
       '2013-04-09 00:00:00', '2013-06-09 00:00:00', '2013-08-09 00:00:00',
       '2013-10-09 00:00:00', '2013-12-09 00:00:00', '14/9/2013', '16/9/2013',
       '18/9/2013', '20/9/2013', '22/9/2013', '24/9/2013', '26/9/2013',
       '28/9/2013', '30/9/2013', '2013-02-10 00:00:00', '2013-04-10 00:00:00',
       '2013-06-10 00:00:00', '2010-08-10 00:00:00', '2013-10-10 00:00:00',
       '2013-12-10 00:00:00', 'Unnamed: 24', 'Unnamed: 25', 'Unnamed: 26',
       'Unnamed: 27', 'Unnamed: 28', 'Unnamed: 29', 'Unnamed: 30',
       'Unnamed: 31', 'Unnamed: 32', 'Unnamed: 33', 'Unnamed: 34',
       'Unnamed: 35'],
      dtype='object')

In [52]:
df1['Unnamed: 27'].isnull()

0      True
1      True
2      True
3      True
4      True
       ... 
495    True
496    True
497    True
498    True
499    True
Name: Unnamed: 27, Length: 500, dtype: bool

In [54]:
df.columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'quantity',
       'discount', 'profit', 'shipping_cost', 'order_priority', 'year',
       'category1'],
      dtype='object')

In [55]:
# Customer Name whose has highest Profit
df['profit']==max(df['profit'])

0        False
1        False
2        False
3        False
4        False
         ...  
51285    False
51286    False
51287    False
51288    False
51289    False
Name: profit, Length: 51290, dtype: bool

In [60]:
df[df['profit']==max(df['profit'])]['customer_name']# Customer with Highest Profit

29530    Tamara Chand
Name: customer_name, dtype: object

In [63]:
#Order Recieved from Sweden
len(df[df['country']=='Sweden'])

203

In [65]:
# Ship Cost Greater Than 80 $
df[df['shipping_cost']>80]

order_id  order_date ship_date       ship_mode  \
7       IN-2011-79397    1/3/2011  1/3/2011        Same Day   
8       ID-2011-80230    1/3/2011  1/9/2011  Standard Class   
9        IZ-2011-4680    1/3/2011  1/7/2011  Standard Class   
27     MX-2011-111255    1/4/2011  1/9/2011    Second Class   
43     MX-2011-109267    1/5/2011  1/9/2011  Standard Class   
...               ...         ...       ...             ...   
51154   IN-2014-49206  12/30/2014  1/1/2015    Second Class   
51228  MX-2014-116267  12/31/2014  1/3/2015    Second Class   
51229   IN-2014-43550  12/31/2014  1/1/2015     First Class   
51230    RS-2014-1460  12/31/2014  1/2/2015    Second Class   
51231   IN-2014-30390  12/31/2014  1/3/2015     First Class   

             customer_name    segment            state      country market  \
7              Kean Nguyen  Corporate  New South Wales    Australia   APAC   
8             Ken Lonsdale   Consumer         Auckland  New Zealand   APAC   
9         Lindsay Williams  Corporate           Ninawa         Iraq   EMEA   
27       Russell Applegate   Consumer           Parana       Brazil  LATAM   
43       Jennifer Halladay   Consumer         Veracruz       Mexico  LATAM   
...                    ...        ...              ...          ...    ...   
51154          Toby Ritter   Consumer       Jawa Barat    Indonesia   APAC   
51228           Erica Bern  Corporate        S?o Paulo       Brazil  LATAM   
51229  Marina Lichtenstein  Corporate          Jakarta    Indonesia   APAC   
51230         Peter Bhler   Consumer    Bashkortostan       Russia   EMEA   
51231     Justin Deggeller  Corporate          Bangkok     Thailand   APAC   

               region  ... sub_category  \
7             Oceania  ...   Appliances   
8             Oceania  ...      Copiers   
9                EMEA  ...       Chairs   
27              South  ...    Bookcases   
43              North  ...   Appliances   
...               ...  ...          ...   
51154  Southeast Asia  ...      Storage   
51228           South  ...      Copiers   
51229  Southeast Asia  ...    Bookcases   
51230            EMEA  ...      Copiers   
51231  Southeast Asia  ...       Tables   

                                  product_name  sales quantity discount  \
7                      Hoover Microwave, White    276        1    0.100   
8                  Hewlett Wireless Fax, Laser    912        4    0.400   
9             Novimex Swivel Stool, Set of Two    667        4    0.000   
27                Dania Classic Bookcase, Pine  1,648        6    0.000   
43                         Hoover Stove, Black  3,029        8    0.000   
...                                        ...    ...      ...      ...   
51154              Rogers Shelving, Wire Frame    457        9    0.170   
51228                Canon Wireless Fax, Color  1,264        5    0.002   
51229           Safco Library with Doors, Pine  1,091        3    0.070   
51230           Hewlett Personal Copier, Laser    866        6    0.000   
51231  Lesro Training Table, Adjustable Height  1,049        9    0.570   

          profit  shipping_cost  order_priority  year category1  
7       110.4120         125.32        Critical  2011      Sudh  
8      -319.4640         107.10             Low  2011      Sudh  
9       253.3200          81.26            High  2011      Sudh  
27      609.8400         109.13          Medium  2011      Sudh  
43      999.3600         191.20          Medium  2011      Sudh  
...          ...            ...             ...   ...       ...  
51154    65.9340         101.88            High  2014      Sudh  
51228   301.4660         253.25            High  2014      Sudh  
51229    46.9206         243.11            High  2014      Sudh  
51230    51.8400         138.18            High  2014      Sudh  
51231 -1195.2387          86.86            High  2014      Sudh  

[3984 rows x 22 columns]

In [81]:
# list of country whose shipping Cost >80
df[df['shipping_cost']>80]['country']
# list of country whose shipping Cost >80 wthh description
df[df['shipping_cost']>80]['country'].describe()

count              3984
unique              120
top       United States
freq                701
Name: country, dtype: object

In [67]:
df.head()

order_id order_date ship_date       ship_mode    customer_name  \
0     AG-2011-2040   1/1/2011  1/6/2011  Standard Class  Toby Braunhardt   
1    IN-2011-47883   1/1/2011  1/8/2011  Standard Class      Joseph Holt   
2     HU-2011-1220   1/1/2011  1/5/2011    Second Class    Annie Thurman   
3  IT-2011-3647632   1/1/2011  1/5/2011    Second Class     Eugene Moren   
4    IN-2011-47883   1/1/2011  1/8/2011  Standard Class      Joseph Holt   

       segment            state    country  market   region  ... sub_category  \
0     Consumer      Constantine    Algeria  Africa   Africa  ...      Storage   
1     Consumer  New South Wales  Australia    APAC  Oceania  ...     Supplies   
2     Consumer         Budapest    Hungary    EMEA     EMEA  ...      Storage   
3  Home Office        Stockholm     Sweden      EU    North  ...        Paper   
4     Consumer  New South Wales  Australia    APAC  Oceania  ...  Furnishings   

                  product_name sales quantity discount   profit  \
0          Tenex Lockers, Blue   408        2      0.0  106.140   
1     Acme Trimmer, High Speed   120        3      0.1   36.036   
2      Tenex Box, Single Width    66        4      0.0   29.640   
3  Enermax Note Cards, Premium    45        3      0.5  -26.055   
4   Eldon Light Bulb, Duo Pack   114        5      0.1   37.770   

   shipping_cost  order_priority  year category1  
0          35.46          Medium  2011      Sudh  
1           9.72          Medium  2011      Sudh  
2           8.17            High  2011      Sudh  
3           4.82            High  2011      Sudh  
4           4.70          Medium  2011      Sudh  

[5 rows x 22 columns]

In [76]:
# shipping>100$ and Prift less than 10 $
df[(df['shipping_cost'] > 100) & (df['profit'] < 10)]

order_id  order_date   ship_date       ship_mode  \
8        ID-2011-80230    1/3/2011    1/9/2011  Standard Class   
257     MX-2011-146584   1/21/2011   1/27/2011  Standard Class   
451    ES-2011-1640672    2/2/2011    2/5/2011     First Class   
625      ID-2011-47575   2/15/2011   2/21/2011  Standard Class   
626      ID-2011-47575   2/15/2011   2/21/2011  Standard Class   
...                ...         ...         ...             ...   
50781   CA-2014-147956  12/25/2014    1/1/2015  Standard Class   
50969   US-2014-120166  12/27/2014  12/31/2014  Standard Class   
50970  ES-2014-4296634  12/27/2014    1/2/2015  Standard Class   
51034    ID-2014-68183  12/29/2014  12/31/2014     First Class   
51153    ID-2014-51509  12/30/2014    1/1/2015    Second Class   

              customer_name    segment                         state  \
8              Ken Lonsdale   Consumer                      Auckland   
257            Hunter Lopez   Consumer                       Managua   
451              Ken Heidel  Corporate                        Saxony   
625    Christopher Martinez   Consumer              National Capital   
626    Christopher Martinez   Consumer              National Capital   
...                     ...        ...                           ...   
50781            Alan Hwang   Consumer                    Washington   
50969          Frank Hawley  Corporate               Valle del Cauca   
50970           Harry Marie  Corporate                      Brittany   
51034             Adam Hart  Corporate                      Victoria   
51153          Naresj Patel   Consumer  Australian Capital Territory   

             country market          region  ... sub_category  \
8        New Zealand   APAC         Oceania  ...      Copiers   
257        Nicaragua  LATAM         Central  ...       Tables   
451          Germany     EU         Central  ...       Phones   
625      Philippines   APAC  Southeast Asia  ...       Phones   
626      Philippines   APAC  Southeast Asia  ...      Copiers   
...              ...    ...             ...  ...          ...   
50781  United States     US            West  ...      Storage   
50969       Colombia  LATAM           South  ...       Tables   
50970         France     EU         Central  ...   Appliances   
51034      Australia   APAC         Oceania  ...   Appliances   
51153      Australia   APAC         Oceania  ...       Chairs   

                                            product_name  sales quantity  \
8                            Hewlett Wireless Fax, Laser    912        4   
257         Chromcraft Conference Table, Fully Assembled  2,797        6   
451                    Motorola Signal Booster, Cordless    584        8   
625                 Nokia Signal Booster, with Caller ID    921        9   
626                      Brother Fax Machine, High-Speed  1,235        6   
...                                                  ...    ...      ...   
50781  Office Impressions Heavy Duty Welded Shelving ...  1,004        6   
50969               Barricks Training Table, Rectangular    842        5   
50970                    Hamilton Beach Microwave, White  1,010        4   
51034                                  Hoover Stove, Red  1,535        3   
51153             Harbour Creations Rocking Chair, Black    434        5   

      discount   profit  shipping_cost  order_priority  year category1  
8         0.40 -319.464         107.10             Low  2011      Sudh  
257       0.20   -0.072         320.38             Low  2011      Sudh  
451       0.50  -58.560         109.14            High  2011      Sudh  
625       0.25 -172.125         128.01             Low  2011      Sudh  
626       0.35 -456.174         118.63             Low  2011      Sudh  
...        ...      ...            ...             ...   ...       ...  
50781     0.00    0.000         121.83             Low  2014      Sudh  
50969     0.20 -200.080         120.11            High  2014      Sudh  
50970

In [72]:
df.dtypes

order_id           object
order_date         object
ship_date          object
ship_mode          object
customer_name      object
segment            object
state              object
country            object
market             object
region             object
product_id         object
category           object
sub_category       object
product_name       object
sales              object
quantity            int64
discount          float64
profit            float64
shipping_cost     float64
order_priority     object
year                int64
category1          object
dtype: object

In [84]:
# Date Fromat
date=pd.to_datetime(df['order_date'])

In [85]:
date

0       2011-01-01
1       2011-01-01
2       2011-01-01
3       2011-01-01
4       2011-01-01
           ...    
51285   2014-12-31
51286   2014-12-31
51287   2014-12-31
51288   2014-12-31
51289   2014-12-31
Name: order_date, Length: 51290, dtype: datetime64[ns]

In [87]:
type(date[0])

pandas._libs.tslibs.timestamps.Timestamp

In [88]:
df['convert_order_date']=pd.to_datetime(df['order_date'])

In [89]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...                                     product_name sales  \
0       Africa  ...                              Tenex Lockers, Blue   408   
1      Oceania  ...                         Acme Trimmer, High Speed   120   
2         EMEA  ...                          Tenex Box, Single Width    66   
3        North  ...                      Enermax Note Cards, Premium    45   
4      Oceania  ...                       Eldon Light Bulb, Duo Pack   114   
...        ...  ...                                              ...   ...   
51285     West  ...  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14   
51286   Africa  ...          Wilson Jones Hole Reinforcements, Clear     4   
51287  Central  ...           Hon Color Coded Labels, 5000 Label Set    26   
51288    North  ...           Hon Legal Exhibit Labels, Alphabetical     7   
51289     West  ...                              Bagged Rubber Bands     3   

      quantity discount    profit  shipping_cost  order_priority  year  \
0            2      0.0  106.1400          35.46          Medium  2011   
1            3      0.1   36.0360           9.72          Medium  2011   
2            4      0.0   29.6400           8.17            High  2011   
3            3      0.5  -26.0550           4.82            High  2011   
4            5      0.1   37.7700           4.70          Medium  2011   
...        ...      ...       ...            ...             ...   ...   
51285        2      0.2    4.5188           0.89          Medium  2014   
51286        1      0.0    0.4200           0.49          Medium  2014   
51287        3      0.0   12.3600           0.35          Medium  2014   
51288        1      0.0    0.5600           0.20          Medium  2014   
51289        3      0.2   -0.6048           0.17          Medium  2014   

       category1 convert_order_date  
0           Sudh         2011-01-01  
1           Sudh         2011-01-01  
2           Sudh         2011-01-01  
3           Sudh         2011-01-01  
4           Sudh         2011-01-01  
...          ...                ...  
51285       Sudh         2014-12-31  
51286       Sudh         2014-12-31  
51287       Sudh         2014-12-31  
51288       Sudh         2014-12-31  
51289       Sudh         201

In [90]:
# Extract Year and add Year olumn from converted Order Date
df['order_date_year']=df['convert_order_date'].dt.year

In [91]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... sales quantity discount    profit shipping_cost  \
0       Africa  ...   408        2      0.0  106.1400         35.46   
1      Oceania  ...   120        3      0.1   36.0360          9.72   
2         EMEA  ...    66        4      0.0   29.6400          8.17   
3        North  ...    45        3      0.5  -26.0550          4.82   
4      Oceania  ...   114        5      0.1   37.7700          4.70   
...        ...  ...   ...      ...      ...       ...           ...   
51285     West  ...    14        2      0.2    4.5188          0.89   
51286   Africa  ...     4        1      0.0    0.4200          0.49   
51287  Central  ...    26        3      0.0   12.3600          0.35   
51288    North  ...     7        1      0.0    0.5600          0.20   
51289     West  ...     3        3      0.2   -0.6048          0.17   

       order_priority  year  category1  convert_order_date order_date_year  
0              Medium  2011       Sudh          2011-01-01            2011  
1              Medium  2011       Sudh          2011-01-01            2011  
2                High  2011       Sudh          2011-01-01            2011  
3                High  2011       Sudh          2011-01-01            2011  
4              Medium  2011       Sudh          2011-01-01            2011  
...               ...   ...        ...                 ...             ...  
51285          Medium  2014       Sudh          2014-12-31            2014  
51286          Medium  2014       Sudh          2014-12-31            2014  
51287          Medium  2014       Sudh          2014-12-31            2014  
51288          Medium  2014       Sudh          2014-12-31            2014  
51289          Medium  2014       Sudh          2014-12-31            2014  

[51290 rows x 24 columns]

In [92]:
# month Column
df['month_col']=df['convert_order_date'].dt.month

In [99]:
df['week_col']=df['convert_order_date'].dt.weekofyear

<ipython-input-99-50adc99c8d79>:1: FutureWarning: Series.dt.weekofyear and Series.dt.week have been deprecated.  Please use Series.dt.isocalendar().week instead.
  df['week_col']=df['convert_order_date'].dt.weekofyear


In [100]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
1      Oceania  ...      0.1   36.0360          9.72         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
1           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51290 rows x 26 columns]

In [101]:

df[['week_col','convert_order_date']]

week_col convert_order_date
0            52         2011-01-01
1            52         2011-01-01
2            52         2011-01-01
3            52         2011-01-01
4            52         2011-01-01
...         ...                ...
51285         1         2014-12-31
51286         1         2014-12-31
51287         1         2014-12-31
51288         1         2014-12-31
51289         1         2014-12-31

[51290 rows x 2 columns]

In [102]:
df[df['order_date_year']==2014]

order_id  order_date ship_date       ship_mode  \
33759  ES-2014-2774938    1/1/2014  1/3/2014    Second Class   
33760    IN-2014-47134    1/1/2014  1/5/2014  Standard Class   
33761  IT-2014-4810734    1/1/2014  1/3/2014    Second Class   
33762   US-2014-105830    1/1/2014  1/3/2014     First Class   
33763   CA-2014-160395    1/1/2014  1/7/2014  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name    segment             state        country  market  \
33759       Fred Harton   Consumer     Ile-de-France         France      EU   
33760       Kean Nguyen  Corporate          Victoria      Australia    APAC   
33761      Alan Shonely   Consumer     Ile-de-France         France      EU   
33762      Duane Benoit   Consumer              Ohio  United States      US   
33763     Kelly Lampkin  Corporate            Nevada  United States      US   
...                 ...        ...               ...            ...     ...   
51285        Erica Bern  Corporate        California  United States      US   
51286         Liz Preis   Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton   Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen   Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias   Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
33759  Central  ...     0.15 -111.6225        104.69           High  2014   
33760  Oceania  ...     0.10   46.9980         46.20         Medium  2014   
33761  Central  ...     0.15  -43.1370         37.85       Critical  2014   
33762     East  ...     0.20  -35.2152         30.66           High  2014   
33763     West  ...     0.20   53.9217         26.90            Low  2014   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...     0.20    4.5188          0.89         Medium  2014   
51286   Africa  ...     0.00    0.4200          0.49         Medium  2014   
51287  Central  ...     0.00   12.3600          0.35         Medium  2014   
51288    North  ...     0.00    0.5600          0.20         Medium  2014   
51289     West  ...     0.20   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
33759       Sudh          2014-01-01             2014          1        1  
33760       Sudh          2014-01-01             2014          1        1  
33761       Sudh          2014-01-01             2014          1        1  
33762       Sudh          2014-01-01             2014          1        1  
33763       Sudh          2014-01-01             2014          1        1  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[17531 rows x 26 columns]

In [104]:
# Which Month We are doing Max Sale
df["month_col"].value_counts()

11    6342
12    6302
9     5975
6     5331
8     5078
10    4490
5     3747
7     3166
4     3057
3     3035
1     2599
2     2168
Name: month_col, dtype: int64

In [105]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
1      Oceania  ...      0.1   36.0360          9.72         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
1           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51290 rows x 26 columns]

In [106]:
df.head()

order_id order_date ship_date       ship_mode    customer_name  \
0     AG-2011-2040   1/1/2011  1/6/2011  Standard Class  Toby Braunhardt   
1    IN-2011-47883   1/1/2011  1/8/2011  Standard Class      Joseph Holt   
2     HU-2011-1220   1/1/2011  1/5/2011    Second Class    Annie Thurman   
3  IT-2011-3647632   1/1/2011  1/5/2011    Second Class     Eugene Moren   
4    IN-2011-47883   1/1/2011  1/8/2011  Standard Class      Joseph Holt   

       segment            state    country  market   region  ... discount  \
0     Consumer      Constantine    Algeria  Africa   Africa  ...      0.0   
1     Consumer  New South Wales  Australia    APAC  Oceania  ...      0.1   
2     Consumer         Budapest    Hungary    EMEA     EMEA  ...      0.0   
3  Home Office        Stockholm     Sweden      EU    North  ...      0.5   
4     Consumer  New South Wales  Australia    APAC  Oceania  ...      0.1   

    profit shipping_cost order_priority  year  category1  convert_order_date  \
0  106.140         35.46         Medium  2011       Sudh          2011-01-01   
1   36.036          9.72         Medium  2011       Sudh          2011-01-01   
2   29.640          8.17           High  2011       Sudh          2011-01-01   
3  -26.055          4.82           High  2011       Sudh          2011-01-01   
4   37.770          4.70         Medium  2011       Sudh          2011-01-01   

   order_date_year  month_col week_col  
0             2011          1       52  
1             2011          1       52  
2             2011          1       52  
3             2011          1       52  
4             2011          1       52  

[5 rows x 26 columns]

In [109]:
df.columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'quantity',
       'discount', 'profit', 'shipping_cost', 'order_priority', 'year',
       'category1', 'convert_order_date', 'order_date_year', 'month_col',
       'week_col'],
      dtype='object')

In [119]:
df['Cost_To_Company']=df['discount'] + df['shipping_cost']

In [121]:
df[df['Cost_To_Company'] == max(df['Cost_To_Company'])]

Empty DataFrame
Columns: [order_id, order_date, ship_date, ship_mode, customer_name, segment, state, country, market, region, product_id, category, sub_category, product_name, sales, quantity, discount, profit, shipping_cost, order_priority, year, category1, convert_order_date, order_date_year, month_col, week_col, Cost_To_Company]
Index: []

[0 rows x 27 columns]

In [124]:
df.drop('Cost_To_Company',axis=1)

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
1      Oceania  ...      0.1   36.0360          9.72         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
1           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51290 rows x 26 columns]

In [125]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...    profit shipping_cost order_priority  year category1  \
0       Africa  ...  106.1400         35.46         Medium  2011      Sudh   
1      Oceania  ...   36.0360          9.72         Medium  2011      Sudh   
2         EMEA  ...   29.6400          8.17           High  2011      Sudh   
3        North  ...  -26.0550          4.82           High  2011      Sudh   
4      Oceania  ...   37.7700          4.70         Medium  2011      Sudh   
...        ...  ...       ...           ...            ...   ...       ...   
51285     West  ...    4.5188          0.89         Medium  2014      Sudh   
51286   Africa  ...    0.4200          0.49         Medium  2014      Sudh   
51287  Central  ...   12.3600          0.35         Medium  2014      Sudh   
51288    North  ...    0.5600          0.20         Medium  2014      Sudh   
51289     West  ...   -0.6048          0.17         Medium  2014      Sudh   

       convert_order_date  order_date_year  month_col  week_col  \
0              2011-01-01             2011          1        52   
1              2011-01-01             2011          1        52   
2              2011-01-01             2011          1        52   
3              2011-01-01             2011          1        52   
4              2011-01-01             2011          1        52   
...                   ...              ...        ...       ...   
51285          2014-12-31             2014         12         1   
51286          2014-12-31             2014         12         1   
51287          2014-12-31             2014         12         1   
51288          2014-12-31             2014         12         1   
51289          2014-12-31             2014         12         1   

      Cost_To_Company  
0               35.46  
1                9.82  
2                8.17  
3                5.32  
4                4.80  
...               ...  
51285            1.09  
51286            0.49  
51287            0.35  
51288            0.20  
51289            0.37  

[51290 rows x 27 columns]

In [126]:
df.head()

order_id order_date ship_date       ship_mode    customer_name  \
0     AG-2011-2040   1/1/2011  1/6/2011  Standard Class  Toby Braunhardt   
1    IN-2011-47883   1/1/2011  1/8/2011  Standard Class      Joseph Holt   
2     HU-2011-1220   1/1/2011  1/5/2011    Second Class    Annie Thurman   
3  IT-2011-3647632   1/1/2011  1/5/2011    Second Class     Eugene Moren   
4    IN-2011-47883   1/1/2011  1/8/2011  Standard Class      Joseph Holt   

       segment            state    country  market   region  ...   profit  \
0     Consumer      Constantine    Algeria  Africa   Africa  ...  106.140   
1     Consumer  New South Wales  Australia    APAC  Oceania  ...   36.036   
2     Consumer         Budapest    Hungary    EMEA     EMEA  ...   29.640   
3  Home Office        Stockholm     Sweden      EU    North  ...  -26.055   
4     Consumer  New South Wales  Australia    APAC  Oceania  ...   37.770   

  shipping_cost order_priority  year category1  convert_order_date  \
0         35.46         Medium  2011      Sudh          2011-01-01   
1          9.72         Medium  2011      Sudh          2011-01-01   
2          8.17           High  2011      Sudh          2011-01-01   
3          4.82           High  2011      Sudh          2011-01-01   
4          4.70         Medium  2011      Sudh          2011-01-01   

   order_date_year  month_col  week_col Cost_To_Company  
0             2011          1        52           35.46  
1             2011          1        52            9.82  
2             2011          1        52            8.17  
3             2011          1        52            5.32  
4             2011          1        52            4.80  

[5 rows x 27 columns]

In [127]:
df.drop('Cost_To_Company',axis=1,inplace=True)

In [128]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
1      Oceania  ...      0.1   36.0360          9.72         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
1           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51290 rows x 26 columns]

In [131]:
df.drop(1)

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
5        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
5           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
5      Oceania  ...      0.1   15.3420          1.80         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
5           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51289 rows x 26 columns]

In [134]:
# Rows Operations
# Loc
#iloc

# Rows 2 and 3 Filter out
df.loc[[2,3]]

order_id order_date ship_date     ship_mode  customer_name  \
2     HU-2011-1220   1/1/2011  1/5/2011  Second Class  Annie Thurman   
3  IT-2011-3647632   1/1/2011  1/5/2011  Second Class   Eugene Moren   

       segment      state  country market region  ... discount  profit  \
2     Consumer   Budapest  Hungary   EMEA   EMEA  ...      0.0  29.640   
3  Home Office  Stockholm   Sweden     EU  North  ...      0.5 -26.055   

  shipping_cost order_priority  year  category1  convert_order_date  \
2          8.17           High  2011       Sudh          2011-01-01   
3          4.82           High  2011       Sudh          2011-01-01   

   order_date_year  month_col week_col  
2             2011          1       52  
3             2011          1       52  

[2 rows x 26 columns]

In [138]:
df[2:10]

order_id order_date ship_date       ship_mode     customer_name  \
2     HU-2011-1220   1/1/2011  1/5/2011    Second Class     Annie Thurman   
3  IT-2011-3647632   1/1/2011  1/5/2011    Second Class      Eugene Moren   
4    IN-2011-47883   1/1/2011  1/8/2011  Standard Class       Joseph Holt   
5    IN-2011-47883   1/1/2011  1/8/2011  Standard Class       Joseph Holt   
6     CA-2011-1510   1/2/2011  1/6/2011  Standard Class   Magdelene Morse   
7    IN-2011-79397   1/3/2011  1/3/2011        Same Day       Kean Nguyen   
8    ID-2011-80230   1/3/2011  1/9/2011  Standard Class      Ken Lonsdale   
9     IZ-2011-4680   1/3/2011  1/7/2011  Standard Class  Lindsay Williams   

       segment            state      country  market   region  ... discount  \
2     Consumer         Budapest      Hungary    EMEA     EMEA  ...      0.0   
3  Home Office        Stockholm       Sweden      EU    North  ...      0.5   
4     Consumer  New South Wales    Australia    APAC  Oceania  ...      0.1   
5     Consumer  New South Wales    Australia    APAC  Oceania  ...      0.1   
6     Consumer          Ontario       Canada  Canada   Canada  ...      0.0   
7    Corporate  New South Wales    Australia    APAC  Oceania  ...      0.1   
8     Consumer         Auckland  New Zealand    APAC  Oceania  ...      0.4   
9    Corporate           Ninawa         Iraq    EMEA     EMEA  ...      0.0   

    profit shipping_cost order_priority  year  category1  convert_order_date  \
2   29.640          8.17           High  2011       Sudh          2011-01-01   
3  -26.055          4.82           High  2011       Sudh          2011-01-01   
4   37.770          4.70         Medium  2011       Sudh          2011-01-01   
5   15.342          1.80         Medium  2011       Sudh          2011-01-01   
6    3.120         24.10         Medium  2011       Sudh          2011-01-02   
7  110.412        125.32       Critical  2011       Sudh          2011-01-03   
8 -319.464        107.10            Low  2011       Sudh          2011-01-03   
9  253.320         81.26           High  2011       Sudh          2011-01-03   

   order_date_year  month_col week_col  
2             2011          1       52  
3             2011          1       52  
4             2011          1       52  
5             2011          1       52  
6             2011          1       52  
7             2011          1        1  
8             2011          1        1  
9             2011          1        1  

[8 rows x 26 columns]

In [139]:
df.loc[2:20:3]

order_id order_date ship_date       ship_mode  customer_name  \
2    HU-2011-1220   1/1/2011  1/5/2011    Second Class  Annie Thurman   
5   IN-2011-47883   1/1/2011  1/8/2011  Standard Class    Joseph Holt   
8   ID-2011-80230   1/3/2011  1/9/2011  Standard Class   Ken Lonsdale   
11  IN-2011-65159   1/3/2011  1/7/2011    Second Class   Larry Blacks   
14  ID-2011-80230   1/3/2011  1/9/2011  Standard Class   Ken Lonsdale   
17  ID-2011-80230   1/3/2011  1/9/2011  Standard Class   Ken Lonsdale   
20  IN-2011-79397   1/3/2011  1/3/2011        Same Day    Kean Nguyen   

      segment             state      country market          region  ...  \
2    Consumer          Budapest      Hungary   EMEA            EMEA  ...   
5    Consumer   New South Wales    Australia   APAC         Oceania  ...   
8    Consumer          Auckland  New Zealand   APAC         Oceania  ...   
11   Consumer  National Capital  Philippines   APAC  Southeast Asia  ...   
14   Consumer          Auckland  New Zealand   APAC         Oceania  ...   
17   Consumer          Auckland  New Zealand   APAC         Oceania  ...   
20  Corporate   New South Wales    Australia   APAC         Oceania  ...   

   discount    profit shipping_cost order_priority  year  category1  \
2      0.00   29.6400          8.17           High  2011       Sudh   
5      0.10   15.3420          1.80         Medium  2011       Sudh   
8      0.40 -319.4640        107.10            Low  2011       Sudh   
11     0.55  -70.3995         21.32           High  2011       Sudh   
14     0.40  -95.6760         10.07            Low  2011       Sudh   
17     0.40    3.4200          8.34            Low  2011       Sudh   
20     0.10    4.7430          7.46       Critical  2011       Sudh   

    convert_order_date  order_date_year  month_col week_col  
2           2011-01-01             2011          1       52  
5           2011-01-01             2011          1       52  
8           2011-01-03             2011          1        1  
11          2011-01-03             2011          1        1  
14          2011-01-03             2011          1        1  
17          2011-01-03             2011          1        1  
20          2011-01-03             2011          1        1  

[7 rows x 26 columns]

In [145]:
# First 4 Rows and first three columns using loc
df.loc[0:4,['order_id','order_date','ship_date']]

order_id order_date ship_date
0     AG-2011-2040   1/1/2011  1/6/2011
1    IN-2011-47883   1/1/2011  1/8/2011
2     HU-2011-1220   1/1/2011  1/5/2011
3  IT-2011-3647632   1/1/2011  1/5/2011
4    IN-2011-47883   1/1/2011  1/8/2011

In [146]:
# First 4 Rows and first three columns using iloc
df.iloc[0:4,0:4]

order_id order_date ship_date       ship_mode
0     AG-2011-2040   1/1/2011  1/6/2011  Standard Class
1    IN-2011-47883   1/1/2011  1/8/2011  Standard Class
2     HU-2011-1220   1/1/2011  1/5/2011    Second Class
3  IT-2011-3647632   1/1/2011  1/5/2011    Second Class

In [147]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
1      Oceania  ...      0.1   36.0360          9.72         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
1           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51290 rows x 26 columns]

In [149]:
df.iloc[1:6,3:6]

ship_mode  customer_name      segment
1  Standard Class    Joseph Holt     Consumer
2    Second Class  Annie Thurman     Consumer
3    Second Class   Eugene Moren  Home Office
4  Standard Class    Joseph Holt     Consumer
5  Standard Class    Joseph Holt     Consumer

In [150]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
1      Oceania  ...      0.1   36.0360          9.72         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
1           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51290 rows x 26 columns]

In [153]:
df[df.dtypes[(df.dtypes=='float64') | (df.dtypes=='int64')].index]

quantity  discount    profit  shipping_cost  year  order_date_year  \
0             2       0.0  106.1400          35.46  2011             2011   
1             3       0.1   36.0360           9.72  2011             2011   
2             4       0.0   29.6400           8.17  2011             2011   
3             3       0.5  -26.0550           4.82  2011             2011   
4             5       0.1   37.7700           4.70  2011             2011   
...         ...       ...       ...            ...   ...              ...   
51285         2       0.2    4.5188           0.89  2014             2014   
51286         1       0.0    0.4200           0.49  2014             2014   
51287         3       0.0   12.3600           0.35  2014             2014   
51288         1       0.0    0.5600           0.20  2014             2014   
51289         3       0.2   -0.6048           0.17  2014             2014   

       month_col  week_col  
0              1        52  
1              1        52  
2              1        52  
3              1        52  
4              1        52  
...          ...       ...  
51285         12         1  
51286         12         1  
51287         12         1  
51288         12         1  
51289         12         1  

[51290 rows x 8 columns]

In [154]:
df2=df[df.dtypes[(df.dtypes=='float64') | (df.dtypes=='int64')].index]

In [155]:
df2[df2>100]

quantity  discount  profit  shipping_cost  year  order_date_year  \
0           NaN       NaN  106.14            NaN  2011             2011   
1           NaN       NaN     NaN            NaN  2011             2011   
2           NaN       NaN     NaN            NaN  2011             2011   
3           NaN       NaN     NaN            NaN  2011             2011   
4           NaN       NaN     NaN            NaN  2011             2011   
...         ...       ...     ...            ...   ...              ...   
51285       NaN       NaN     NaN            NaN  2014             2014   
51286       NaN       NaN     NaN            NaN  2014             2014   
51287       NaN       NaN     NaN            NaN  2014             2014   
51288       NaN       NaN     NaN            NaN  2014             2014   
51289       NaN       NaN     NaN            NaN  2014             2014   

       month_col  week_col  
0            NaN       NaN  
1            NaN       NaN  
2            NaN       NaN  
3            NaN       NaN  
4            NaN       NaN  
...          ...       ...  
51285        NaN       NaN  
51286        NaN       NaN  
51287        NaN       NaN  
51288        NaN       NaN  
51289        NaN       NaN  

[51290 rows x 8 columns]

In [156]:
df3=df2[df2>100]

In [158]:
# Drop The Rows and columns having null values on based on Rows and columns
df3.dropna()

Empty DataFrame
Columns: [quantity, discount, profit, shipping_cost, year, order_date_year, month_col, week_col]
Index: []

In [159]:
df3

quantity  discount  profit  shipping_cost  year  order_date_year  \
0           NaN       NaN  106.14            NaN  2011             2011   
1           NaN       NaN     NaN            NaN  2011             2011   
2           NaN       NaN     NaN            NaN  2011             2011   
3           NaN       NaN     NaN            NaN  2011             2011   
4           NaN       NaN     NaN            NaN  2011             2011   
...         ...       ...     ...            ...   ...              ...   
51285       NaN       NaN     NaN            NaN  2014             2014   
51286       NaN       NaN     NaN            NaN  2014             2014   
51287       NaN       NaN     NaN            NaN  2014             2014   
51288       NaN       NaN     NaN            NaN  2014             2014   
51289       NaN       NaN     NaN            NaN  2014             2014   

       month_col  week_col  
0            NaN       NaN  
1            NaN       NaN  
2            NaN       NaN  
3            NaN       NaN  
4            NaN       NaN  
...          ...       ...  
51285        NaN       NaN  
51286        NaN       NaN  
51287        NaN       NaN  
51288        NaN       NaN  
51289        NaN       NaN  

[51290 rows x 8 columns]

In [160]:
df3.dropna(axis=1)# Using columns we use axis =1 default it use 

year  order_date_year
0      2011             2011
1      2011             2011
2      2011             2011
3      2011             2011
4      2011             2011
...     ...              ...
51285  2014             2014
51286  2014             2014
51287  2014             2014
51288  2014             2014
51289  2014             2014

[51290 rows x 2 columns]

In [162]:
# More than 5 Na Value Oly then Drop the column or Row 
# For Rows 
df3.dropna(thresh=4)# minimum 5 Nana Value we have it will not delete if 4 then it wil delete 

quantity  discount    profit  shipping_cost  year  order_date_year  \
7           NaN       NaN  110.4120         125.32  2011             2011   
27          NaN       NaN  609.8400         109.13  2011             2011   
43          NaN       NaN  999.3600         191.20  2011             2011   
58          NaN       NaN  746.4078         258.99  2011             2011   
84          NaN       NaN  260.8200         275.52  2011             2011   
...         ...       ...       ...            ...   ...              ...   
51149       NaN       NaN  899.1000         236.49  2014             2014   
51150       NaN       NaN  317.1600         184.60  2014             2014   
51151       NaN       NaN  768.8800         155.85  2014             2014   
51152       NaN       NaN  314.0384         139.47  2014             2014   
51228       NaN       NaN  301.4660         253.25  2014             2014   

       month_col  week_col  
7            NaN       NaN  
27           NaN       NaN  
43           NaN       NaN  
58           NaN       NaN  
84           NaN       NaN  
...          ...       ...  
51149        NaN       NaN  
51150        NaN       NaN  
51151        NaN       NaN  
51152        NaN       NaN  
51228        NaN       NaN  

[1941 rows x 8 columns]

In [164]:
# Fill Nan values with Some Value 
df3.fillna(value=4)

quantity  discount  profit  shipping_cost  year  order_date_year  \
0           4.0       4.0  106.14            4.0  2011             2011   
1           4.0       4.0    4.00            4.0  2011             2011   
2           4.0       4.0    4.00            4.0  2011             2011   
3           4.0       4.0    4.00            4.0  2011             2011   
4           4.0       4.0    4.00            4.0  2011             2011   
...         ...       ...     ...            ...   ...              ...   
51285       4.0       4.0    4.00            4.0  2014             2014   
51286       4.0       4.0    4.00            4.0  2014             2014   
51287       4.0       4.0    4.00            4.0  2014             2014   
51288       4.0       4.0    4.00            4.0  2014             2014   
51289       4.0       4.0    4.00            4.0  2014             2014   

       month_col  week_col  
0            4.0       4.0  
1            4.0       4.0  
2            4.0       4.0  
3            4.0       4.0  
4            4.0       4.0  
...          ...       ...  
51285        4.0       4.0  
51286        4.0       4.0  
51287        4.0       4.0  
51288        4.0       4.0  
51289        4.0       4.0  

[51290 rows x 8 columns]

In [165]:
# Fill a Data For A column with Mean of Profit
df3.fillna(value=df3['profit'].mean())

quantity    discount      profit  shipping_cost  year  \
0      285.710477  285.710477  106.140000     285.710477  2011   
1      285.710477  285.710477  285.710477     285.710477  2011   
2      285.710477  285.710477  285.710477     285.710477  2011   
3      285.710477  285.710477  285.710477     285.710477  2011   
4      285.710477  285.710477  285.710477     285.710477  2011   
...           ...         ...         ...            ...   ...   
51285  285.710477  285.710477  285.710477     285.710477  2014   
51286  285.710477  285.710477  285.710477     285.710477  2014   
51287  285.710477  285.710477  285.710477     285.710477  2014   
51288  285.710477  285.710477  285.710477     285.710477  2014   
51289  285.710477  285.710477  285.710477     285.710477  2014   

       order_date_year   month_col    week_col  
0                 2011  285.710477  285.710477  
1                 2011  285.710477  285.710477  
2                 2011  285.710477  285.710477  
3                 2011  285.710477  285.710477  
4                 2011  285.710477  285.710477  
...                ...         ...         ...  
51285             2014  285.710477  285.710477  
51286             2014  285.710477  285.710477  
51287             2014  285.710477  285.710477  
51288             2014  285.710477  285.710477  
51289             2014  285.710477  285.710477  

[51290 rows x 8 columns]

In [166]:
df

order_id  order_date ship_date       ship_mode  \
0         AG-2011-2040    1/1/2011  1/6/2011  Standard Class   
1        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
2         HU-2011-1220    1/1/2011  1/5/2011    Second Class   
3      IT-2011-3647632    1/1/2011  1/5/2011    Second Class   
4        IN-2011-47883    1/1/2011  1/8/2011  Standard Class   
...                ...         ...       ...             ...   
51285   CA-2014-115427  12/31/2014  1/4/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  1/5/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  1/2/2015    Second Class   
51288   MX-2014-114783  12/31/2014  1/6/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  1/4/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ... discount    profit shipping_cost order_priority  year  \
0       Africa  ...      0.0  106.1400         35.46         Medium  2011   
1      Oceania  ...      0.1   36.0360          9.72         Medium  2011   
2         EMEA  ...      0.0   29.6400          8.17           High  2011   
3        North  ...      0.5  -26.0550          4.82           High  2011   
4      Oceania  ...      0.1   37.7700          4.70         Medium  2011   
...        ...  ...      ...       ...           ...            ...   ...   
51285     West  ...      0.2    4.5188          0.89         Medium  2014   
51286   Africa  ...      0.0    0.4200          0.49         Medium  2014   
51287  Central  ...      0.0   12.3600          0.35         Medium  2014   
51288    North  ...      0.0    0.5600          0.20         Medium  2014   
51289     West  ...      0.2   -0.6048          0.17         Medium  2014   

       category1  convert_order_date  order_date_year  month_col week_col  
0           Sudh          2011-01-01             2011          1       52  
1           Sudh          2011-01-01             2011          1       52  
2           Sudh          2011-01-01             2011          1       52  
3           Sudh          2011-01-01             2011          1       52  
4           Sudh          2011-01-01             2011          1       52  
...          ...                 ...              ...        ...      ...  
51285       Sudh          2014-12-31             2014         12        1  
51286       Sudh          2014-12-31             2014         12        1  
51287       Sudh          2014-12-31             2014         12        1  
51288       Sudh          2014-12-31             2014         12        1  
51289       Sudh          2014-12-31             2014         12        1  

[51290 rows x 26 columns]

In [167]:
# Average Profit received year Wise
df.groupby('order_date_year')['profit'].mean()

order_date_year
2011    27.666238
2012    28.043722
2013    29.604519
2014    28.758540
Name: profit, dtype: float64

In [169]:
df.groupby('order_date_year')['shipping_cost'].mean()

order_date_year
2011    27.147264
2012    25.861329
2013    26.418577
2014    26.268197
Name: shipping_cost, dtype: float64

In [170]:
df.columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'quantity',
       'discount', 'profit', 'shipping_cost', 'order_priority', 'year',
       'category1', 'convert_order_date', 'order_date_year', 'month_col',
       'week_col'],
      dtype='object')

In [171]:
df[['country', 'market']]

country  market
0            Algeria  Africa
1          Australia    APAC
2            Hungary    EMEA
3             Sweden      EU
4          Australia    APAC
...              ...     ...
51285  United States      US
51286        Morocco  Africa
51287      Nicaragua   LATAM
51288         Mexico   LATAM
51289  United States      US

[51290 rows x 2 columns]

In [172]:
df.groupby('order_date_year')['sales'].mean()

DataError: No numeric types to aggregate

In [175]:
df['sales'].astype(float)

ValueError: could not convert string to float: '1,648'

In [176]:
df['sales']

0        408
1        120
2         66
3         45
4        114
        ... 
51285     14
51286      4
51287     26
51288      7
51289      3
Name: sales, Length: 51290, dtype: object

In [178]:
df.sales.str.replace(',','').astype(int)

0        408
1        120
2         66
3         45
4        114
        ... 
51285     14
51286      4
51287     26
51288      7
51289      3
Name: sales, Length: 51290, dtype: int32

In [179]:
df['sales']=df.sales.str.replace(',','').astype(int)

In [180]:
df.describe()

sales      quantity      discount        profit  shipping_cost  \
count  51290.000000  51290.000000  51290.000000  51290.000000   51290.000000   
mean     246.498440      3.476545      0.142908     28.641740      26.375915   
std      487.567175      2.278766      0.212280    174.424113      57.296804   
min        0.000000      1.000000      0.000000  -6599.978000       0.000000   
25%       31.000000      2.000000      0.000000      0.000000       2.610000   
50%       85.000000      3.000000      0.000000      9.240000       7.790000   
75%      251.000000      5.000000      0.200000     36.810000      24.450000   
max    22638.000000     14.000000      0.850000   8399.976000     933.570000   

               year  order_date_year     month_col      week_col  
count  51290.000000     51290.000000  51290.000000  51290.000000  
mean    2012.777208      2012.777208      7.522636     30.675687  
std        1.098931         1.098931      3.300056     14.535168  
min     2011.000000      2011.000000      1.000000      1.000000  
25%     2012.000000      2012.000000      5.000000     20.000000  
50%     2013.000000      2013.000000      8.000000     33.000000  
75%     2014.000000      2014.000000     10.000000     44.000000  
max     2014.000000      2014.000000     12.000000     52.000000

In [181]:
df.groupby('order_date_year')['sales'].mean()

order_date_year
2011    251.112581
2012    244.252235
2013    246.819335
2014    245.282129
Name: sales, dtype: float64

In [182]:
df.groupby('order_date_year')['quantity'].mean()

order_date_year
2011    3.494443
2012    3.476647
2013    3.488369
2014    3.457989
Name: quantity, dtype: float64

In [183]:
df.groupby('order_date_year')['quantity'].sum()

order_date_year
2011    31443
2012    38111
2013    48136
2014    60622
Name: quantity, dtype: int64

In [184]:
df.groupby('order_date_year')['sales'].sum()

order_date_year
2011    2259511
2012    2677493
2013    3405860
2014    4300041
Name: sales, dtype: int32

# https://drive.google.com/file/d/1FRnbQqgK6S-mr9StNF5MurSaXhah9uoJ/view?usp=sharing


https://drive.google.com/file/d/11rPJXr-9VoeV2Pi0RSfVehw1tbQOAjLI/view?usp=sharing

https://docs.google.com/spreadsheets/d/1z4EWsGxw0YBQ1-51xZDxYNlB3ngJzg4t/edit?usp=sharing&ouid=118282207943964605599&rtpof=true&sd=true


https://docs.google.com/spreadsheets/d/1z2n8aeFF5GoohQ3KCZw8bKogezpO9l4k/edit?usp=sharing&ouid=118282207943964605599&rtpof=true&sd=true



Q1. Each And Every Agent Weekly Report Average Rating on weekly basis
Keep in minf tht they two days for holiday
Q2. Totsl working days for each agent
3.total query you have solved
4.total feedback received
5.Agent Name who have average rating beltwwen 3.5 to 4
6 aGENT NAME WHO HAS RATING LESS THAN 3.5
7. Rating whoo has Agent Name more than 4.5
8.How many fedback Agent have received more than Average
9.Average weekly response time for each Agent
10.
Average weekly Resolution for each Agent
kist of al lagent Name
percentage of Chat they received a feedback
total contrubtuion hour for each anfd every sgents weekly basisi 

Total Percentage of actve houra for a month 